File: src/main/java/com/ai/driven/cybersecurity/platform/notify/EmailNotifier.java
package com.ai.driven.cybersecurity.platform.notify;

import java.util.Properties;

import jakarta.mail.Authenticator;
import jakarta.mail.Message;
import jakarta.mail.PasswordAuthentication;
import jakarta.mail.Session;
import jakarta.mail.Transport;
import jakarta.mail.internet.InternetAddress;
import jakarta.mail.internet.MimeMessage;

public class EmailNotifier {

	private static final String SMTP_HOST = "smtp.mail.yahoo.com";
	private static final String SMTP_PORT = "587"; // TLS
	private static final String SMTP_USER = "sunayanasamanjobsearch@yahoo.com";
	private static final String SMTP_PASSWORD = "mzuywktaofjamcpo"; // Use App Password
	private static final String FROM_EMAIL = SMTP_USER;

	public static void sendEmail(String subject, String body, String toEmail) {
		try {
			Properties props = new Properties();
			props.put("mail.smtp.auth", "true");
			props.put("mail.smtp.starttls.enable", "true");
			props.put("mail.smtp.host", SMTP_HOST);
			props.put("mail.smtp.port", SMTP_PORT);

			Session session = Session.getInstance(props, new Authenticator() {
				protected PasswordAuthentication getPasswordAuthentication() {
					return new PasswordAuthentication(SMTP_USER, SMTP_PASSWORD);
				}
			});

			Message message = new MimeMessage(session);
			message.setFrom(new InternetAddress(FROM_EMAIL));
			message.setRecipients(Message.RecipientType.TO, InternetAddress.parse(toEmail));
			message.setSubject(subject);
			message.setText(body);

			Transport.send(message);
			System.out.println("✅ Email sent successfully to " + toEmail);

		} catch (Exception e) {
			e.printStackTrace(); // Print full stacktrace for debugging
			System.err.println("❌ Error sending email: " + e.getMessage());
		}
	}
}

File: bin/python_ai/notifiers/email_notifier.py
# email_notifier.py
import smtplib
from email.mime.text import MIMEText
from email.mime.multipart import MIMEMultipart

# ----------------- CONFIG -----------------
SMTP_HOST = "smtp.yourmail.com"
SMTP_PORT = 587  # 465 for SSL, 587 for TLS
SMTP_USER = "your_email@domain.com"
SMTP_PASSWORD = "your_email_password"
FROM_EMAIL = "your_email@domain.com"
# ------------------------------------------

def send_email(subject, message, to_email):
    """
    Send Email via SMTP
    :param subject: string, email subject
    :param message: string, email body
    :param to_email: string, recipient email
    """
    try:
        msg = MIMEMultipart()
        msg['From'] = FROM_EMAIL
        msg['To'] = to_email
        msg['Subject'] = subject
        msg.attach(MIMEText(message, 'plain'))

        server = smtplib.SMTP(SMTP_HOST, SMTP_PORT)
        server.starttls()  # Enable TLS
        server.login(SMTP_USER, SMTP_PASSWORD)
        server.send_message(msg)
        server.quit()

        print(f"Email sent successfully to {to_email}")
    except Exception as e:
        print(f"Error sending email: {e}")
